impexp-core/src/main/java/org/citydb/query/builder/config/ConfigQueryBuilder.java

```diff
@@ -317,19 +319,52 @@ public Query buildQuery(SimpleKmlQuery queryConfig, NamespaceContext namespaceCo
 			}
 		}
 
-		if (queryConfig.getMode() == SimpleKmlQueryMode.SINGLE) {
-			// gml:id filter
-			if (queryConfig.isSetGmlIdFilter() && queryConfig.getGmlIdFilter().isSetResourceIds())
-				query.setSelection(new SelectionFilter(predicateBuilder.buildPredicate(queryConfig.getGmlIdFilter())));
-		} else {
-			// bbox filter
-			if (queryConfig.isSetBboxFilter()) {
-				KmlTiling bboxFilter = queryConfig.getBboxFilter();
-				if (!bboxFilter.isSetExtent())
-					throw new QueryBuildException("The bounding box filter requires an " + GeometryType.ENVELOPE + " as spatial operand.");
+		List<Predicate> predicates = new ArrayList<>();
 
-				TilingFilterBuilder tilingFilterBuilder = new TilingFilterBuilder(databaseAdapter);
-				query.setTiling(tilingFilterBuilder.buildTilingFilter(bboxFilter));
+		// gml:id filter
+		if (queryConfig.isUseGmlIdFilter() && queryConfig.isSetGmlIdFilter() && queryConfig.getGmlIdFilter().isSetResourceIds()) {
+			predicates.add(predicateBuilder.buildPredicate(queryConfig.getGmlIdFilter()));
+		}
+
+		// calculate extent if the bbox filter is disabled
+		KmlTiling spatialFilter = queryConfig.getSpatialFilter();
+		if (!queryConfig.isUseBboxFilter()) {
+			try {
+				Query extentQuery = new Query(query);
+				if (!predicates.isEmpty()) {
+					// add gml:id filter to selection
+					extentQuery.setSelection(new SelectionFilter(predicates.get(0)));
+				}
+				spatialFilter.setExtent(databaseAdapter.getUtil().calcBoundingBox(workspace, query, schemaMapping));
+			} catch (SQLException | FilterException e) {
+				throw new QueryBuildException("Failed to calculate bounding box based on the non-spatial filter settings.", e);
+			}
+		}
+
+		// bbox filter
+		if (queryConfig.isUseBboxFilter()) {
+			if (!spatialFilter.isSetExtent())
+				throw new QueryBuildException("The bounding box filter requires an " + GeometryType.ENVELOPE + " as spatial operand.");
+
+			if (spatialFilter.getMode() == KmlTilingMode.NO_TILING) {
+				BBOXOperator bbox = new BBOXOperator();
+				bbox.setEnvelope(spatialFilter.getExtent());
+				predicates.add(predicateBuilder.buildPredicate(bbox));
+			}
+		}
+
+		// apply tiling
+		if (spatialFilter.getMode() != KmlTilingMode.NO_TILING && spatialFilter.isSetExtent()) {
+			TilingFilterBuilder tilingFilterBuilder = new TilingFilterBuilder(databaseAdapter);
+			query.setTiling(tilingFilterBuilder.buildTilingFilter(spatialFilter));
+		}
+
+		if (!predicates.isEmpty()) {
```

Comment:
If neither gml:id, bbox nor tiling is set, then still an extent is calculated in line 338 (because `!queryConfig.isUseBboxFilter()` is true) and the result is assigned to the `spatialFilter` object. But this extent is not used anywhere later, because `spatialFilter.getMode() != KmlTilingMode.NO_TILING` is false in line 357. So, the extent is calculated but never used in this scenario. I think this needs to be adapted here.

What is the intended purpose of the automatic bbox? Should it always be calculated and used whenever the user does not provide a bbox?

Reply:
Since the output master KML and JSON files require the extent of the exported features, a bbox must be either automatically calculated or provided by users.

Reply:
See below